File: app/src/main/java/hu/napirajz/android/HeightWrapBitmapTarget.java
package hu.napirajz.android;

import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;
import android.widget.ImageView;

import com.squareup.picasso.Picasso;
import com.squareup.picasso.Target;

import lombok.AllArgsConstructor;

@AllArgsConstructor(suppressConstructorProperties = true)
public class HeightWrapBitmapTarget implements Target {

    private int width;
    private ImageView imageView;
    private OnFinishListener onFinishListener;

    @Override
    public void onBitmapLoaded(Bitmap bitmap, Picasso.LoadedFrom from) {
        float w = bitmap.getWidth();
        int newHeight;
        Bitmap scaled;
        double ratio = (double) width / w;
        newHeight = (int) (bitmap.getHeight() * ratio);
        scaled = Bitmap.createScaledBitmap(bitmap, width, newHeight, false);

        imageView.setImageBitmap(scaled);
        imageView.getLayoutParams().height = newHeight;
        imageView.getLayoutParams().width = width;

        if (onFinishListener != null) {
            onFinishListener.success(bitmap);
        }
    }

    @Override
    public void onBitmapFailed(Drawable errorDrawable) {
        if (onFinishListener != null) {
            onFinishListener.error();
        }

    }

    @Override
    public void onPrepareLoad(Drawable placeHolderDrawable) {

    }
}

File: app/src/main/kotlin/hu/napirajz/android/HeightWrapBitmapTarget.kt
package hu.napirajz.android

import android.graphics.Bitmap
import android.graphics.drawable.Drawable
import android.view.View
import android.widget.ImageView
import android.widget.ProgressBar

import com.squareup.picasso.Picasso
import com.squareup.picasso.Target

import lombok.AllArgsConstructor

class HeightWrapBitmapTarget(var width: Int, var imageView: ImageView, var progressBar: ProgressBar) : Target {

    override fun onBitmapLoaded(bitmap: Bitmap, from: Picasso.LoadedFrom) {
        val w = bitmap.width.toFloat()
        val newHeight: Int
        val scaled: Bitmap
        val ratio = width.toDouble() / w
        newHeight = (bitmap.height * ratio).toInt()
        scaled = Bitmap.createScaledBitmap(bitmap, width, newHeight, false)

        imageView.setImageBitmap(scaled)
        imageView.layoutParams.height = newHeight
        imageView.layoutParams.width = width

        imageView.visibility = View.VISIBLE
        progressBar.visibility = View.GONE
    }

    override fun onBitmapFailed(errorDrawable: Drawable) {
        imageView.visibility = View.VISIBLE
        progressBar.visibility = View.GONE

    }

    override fun onPrepareLoad(placeHolderDrawable: Drawable) {

    }
}
